flink-table-store-core/src/main/java/org/apache/flink/table/store/CatalogOptions.java

```diff
@@ -49,6 +49,12 @@ private CatalogOptions() {}
                     .noDefaultValue()
                     .withDescription("Uri of metastore server.");
 
+    public static final ConfigOption<Boolean> HMS_EXTERNAL =
+            ConfigOptions.key("hms.external")
```

Comment:
Could this introduce the `TableType` enum and `table.type` config option which default value is `MANAGED_TABLE`? cc @JingsongLi

Reply:
@SteNicholas Thanks for review, updated